docs/development/releases-and-deployment.md

```diff
@@ -0,0 +1,102 @@
+# Spark release / deployment guidelines
+
+## GitHub releases
+
+* https://github.com/Midburn/Spark/releases
+* There should always be 1 draft release against master branch
+* This draft release is updated on each merge or commit to master.
+
+### Publishing a release
+
+* When release is ready (e.g. have enough content / is stable enough)- you can publish it
+* **Before publishing on GitHub** - you should update the [package.json](/package.json) "version" attribute to match the GitHub release you are about to publish
+* merge the change in package.json to master
+* don't forget to push!
+* now, you can publish the release in GitHub
+
+## Manual Deployment
+
```

Comment:
I see you've opted to build-->upload-->run deployment cycle.

A different cycle I've used in the past is to git pull a specific tag on the prod/staging machine.
This allowed us to debug / code on the production/staging systems (not necessarily a good practice, but it did allow for some flexibility).

I don't have issues with the path you've chosen, but perhaps a short discussion/analysis of the design considerations would be good here in the documentation.